extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts

```diff
@@ -566,8 +566,15 @@ export class PublishDatabaseDialog {
 					this.targetDatabaseDropDown!.value = result.databaseName;
 				}
 
-				// save count for later to see if we need to add the SQLCMD Variables table
-				const sqlCmdVarBeforeCount = Object.keys(<Record<string, string>>this.sqlCmdVars).length;
+				if (Object.keys(result.sqlCmdVariables).length) {
+					// add SQLCMD Variables table if it wasn't there before and the profile had sqlcmd variables
```

Comment:
why is the table being added preemptively/before we even check if there are any sqlcmd variables in the project?

Reply:
if the result from reading the publish profile has any sqlcmd variables(line 569), then the table should be added if the project didn't already specify any sqlcmd variables